python/tvm/script/parser.py

```diff
@@ -1138,6 +1141,29 @@ def transform_TypeTuple(self, node):
         """
         return [self.transform(value) for value in node.values]
 
+    def transform_TypeApply(self, node):
+        """Visitor for Type[Type] expressions.
+
+        Mostly used for ``T.Ptr`` expressions.
+        """
+        func = self.transform(node.func_name)
+
+        if not isinstance(func, ty.TypeGeneric):
+            self.report_error(f"Expected a type but found {type(func).__name__}", node.span)
+
+        param_types = []
+        for param in node.params:
+            param_type = self.transform(param)
+            if not isinstance(param_type, ty.TypeGeneric):
```

Comment:
It doesn't, no.  The parameters of `TypeApply` typically aren't `GenericPtrType`.  For example, `T.Ptr[T.int32]`, the `T.int32` parameter is a `ConcreteType`.

At some point, I may see if there's support for renaming `ty.TypeGeneric` to `ty.Type`.  As it is, `ty.ConcreteType` is a subclass of `ty.TypeGeneric`, which doesn't make very much sense to me as there since it require any generic parameters in the user-supplied tvmscript.